### PR TITLE
[Feature]: Add a field for a secret to the Sink Type

### LINKED ISSUE #309: [Feature]: Add a field for a secret to the Sink Type
### Checklist

- [X] I've searched for similar issues and couldn't find anything matching
- [X] I've discussed this feature request in the [K8sGPT Slack](https://k8sgpt.slack.com) and got positive feedback

### Is this feature request related to a problem?

Yes

### Problem Description

I am frustrated trying to add a sink whose endpoint contains an API Key that should remain a secret. I believe there is a security risk associated with saving endpoints that might contain secrets as plaintext strings in the CR, such as the Slack webhook URL, for example.

### Solution Description

I think the best solution would be to add a field in the Sink type for a `secret`. The creator of the sink can then determine how to use this secret to build their request in the `configure` method.

Let's consider a hypothetical Telegram bot as a sink. Requests to the Telegram API are presented in the form `https://api.telegram.org/bot<token>/METHOD_NAME`. The `https://api.telegram.org/bot` could be stored in the endpoint field, and the `token` stored as the secret. The full URL could then be assembled in the `configure` method.

### Benefits

Allows users to create sinks with secrets. Additionally, it allows users to create sinks where authentication is separate from the endpoint. For example, authentication that can be passed via headers instead of being inherently tied to the endpoint.

### Potential Drawbacks

_No response_

### Additional Information

I would be happy to work on this

### PR DESCRIPTION
**Pull request by phillipahereza**
PR is the same as #313 that was accidentally deleted 🤦

Closes #309 

## 📑 Description
This change allows developers to configure the sink with secrets. 
Additionally, it allows developers to create sinks where the authentication is separate from the endpoint.

```yaml
apiVersion: core.k8sgpt.ai/v1alpha1
kind: K8sGPT
metadata:
  name: k8sgpt
  namespace: gpt
spec:
  ai:
    enabled: true
    backend: openai
  sink:
    type: slack
    secret:
      name: slack-webhook
      key: url
```

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [x] All the tests have passed

## ℹ Additional Information

This PR breaks/affects #303

### CODE REVIEW

File: pkg/sinks/slack.go [hunk 1]
```diff
@@ -43,8 +43,12 @@ func buildSlackMessage(kind, name, details, k8sgptCR string) SlackMessage {
 	}
 }
 
-func (s *SlackSink) Configure(config v1alpha1.K8sGPT, c Client) {
-	s.Endpoint = config.Spec.Sink.Endpoint
+func (s *SlackSink) Configure(config v1alpha1.K8sGPT, c Client, secret string) {
+	s.Endpoint = secret
```
Comment (AlexsJones) [inline pkg/sinks/slack.go:47]: Wouldn't this set the endpoint to the secret name rather than a URL?

Comment (phillipahereza) [reply pkg/sinks/slack.go:47]: No, what is passed to the `sink.Configure` is the [secret value](https://github.com/k8sgpt-ai/k8sgpt-operator/blob/d85401b2d6dc8806886c7bba36f173aaa7358f37/controllers/k8sgpt_controller.go#L307). I think I'll change the variable name. It's a bit misleading

**Review by arbreezy — DISMISSED** [review]
Thanks for the contribution @phillipahereza, can you please update Readme to include the new CR spec field and update also the Helm chart's template ?

**Review by arbreezy — APPROVED** [review]

**Review by arbreezy — APPROVED** [review]
I added a minor fix, thanks again for your contribution :) !

**Pull request merged**